xls/contrib/integrator/ir_integrator_test.cc

```diff
@@ -0,0 +1,154 @@
+  fb_body.Literal(UBits(0b11, 2));
+  XLS_ASSERT_OK_AND_ASSIGN(Function * body_func, fb_body.Build());
+
+  FunctionBuilder fb_double("double", p.get());
+  auto double_in = fb_double.Param("in1", p->GetBitsType(2));
+  fb_double.Add(double_in, double_in);
+  XLS_ASSERT_OK_AND_ASSIGN(Function * double_func, fb_double.Build());
+
+  FunctionBuilder fb_main("main", p.get());
+  auto main_in1 = fb_main.Param("in1", p->GetBitsType(2));
+  auto main_in2 = fb_main.Param("in2", p->GetBitsType(2));
+  auto main_in_arr =
+      fb_main.Param("in_arr", p->GetArrayType(2, p->GetBitsType(2)));
+  fb_main.Map(main_in_arr, double_func, /*loc=*/std::nullopt, "map");
+  fb_main.CountedFor(main_in1, /*trip_count=*/4, /*stride=*/1, body_func,
+                     /*invariant_args=*/{}, /*loc=*/std::nullopt,
+                     "counted_for");
+  fb_main.Invoke(/*args=*/{main_in1}, double_func, /*loc=*/std::nullopt,
+                 "invoke");
+  XLS_ASSERT_OK_AND_ASSIGN(Function * main_func, fb_main.Build());
+
+  auto get_function_names = [](Package* p) {
+    std::vector<std::string> names;
+    for (const auto& func : p->functions()) {
+      names.push_back(func->name());
+    }
+    return names;
+  };
+
+  auto get_called_function = [](Package* p) {
+    std::vector<Function*> functions;
+    for (const auto& func : p->functions()) {
+      for (const auto* node : func->nodes()) {
+        if (node->op() == Op::kCountedFor) {
+          functions.push_back(node->As<CountedFor>()->body());
+        }
+        if (node->op() == Op::kInvoke) {
+          functions.push_back(node->As<Invoke>()->to_apply());
+        }
+        if (node->op() == Op::kMap) {
+          functions.push_back(node->As<Map>()->to_apply());
+        }
+      }
+    }
+    return functions;
+  };
+
+  XLS_ASSERT_OK_AND_ASSIGN(std::unique_ptr<IntegrationBuilder> builder,
+                           IntegrationBuilder::Build({main_func}));
+
+  // Original package is unchanged.
+  EXPECT_THAT(get_function_names(p.get()),
+              UnorderedElementsAre("body", "double", "main"));
+
+  // Original CountedFor
+  XLS_ASSERT_OK_AND_ASSIGN(Node * counted_original_generic,
+                           main_func->GetNode("counted_for"));
+  CountedFor* counted_original = counted_original_generic->As<CountedFor>();
+  EXPECT_EQ(counted_original->body(), body_func);
+  EXPECT_EQ(counted_original->body()->name(), "body");
+  EXPECT_EQ(counted_original->body()->package(), p.get());
+
+  // Original Map
+  XLS_ASSERT_OK_AND_ASSIGN(Node * map_original_generic,
+                           main_func->GetNode("map"));
+  Map* map_original = map_original_generic->As<Map>();
+  EXPECT_EQ(map_original->to_apply(), double_func);
+  EXPECT_EQ(map_original->to_apply()->name(), "double");
+  EXPECT_EQ(map_original->to_apply()->package(), p.get());
+
+  // Original Invoke
+  XLS_ASSERT_OK_AND_ASSIGN(Node * invoke_original_generic,
+                           main_func->GetNode("invoke"));
```

Comment:
IrTestBase has FindNode for this purpose:

```
Invoke* invoke_orignal = FindNode("invoke", main_func)->As<Invoke>();
```
It checkfails if it can't find the node but that's fine because it's a test. Saves you from having to unwrap the StatusOr.

Reply:
Done.